src/mds/MDSMap.cc

```diff
@@ -610,37 +611,22 @@ void MDSMap::encode(bufferlist& bl, uint64_t features) const
   }
 
   using ceph::encode;
-  if ((features & CEPH_FEATURE_PGID64) == 0) {
-    __u16 v = 2;
-    encode(v, bl);
-    encode(epoch, bl);
-    encode(flags, bl);
-    encode(last_failure, bl);
-    encode(root, bl);
-    encode(session_timeout, bl);
-    encode(session_autoclose, bl);
-    encode(max_file_size, bl);
-    encode(max_mds, bl);
-    __u32 n = mds_info.size();
-    encode(n, bl);
-    for (map<mds_gid_t, mds_info_t>::const_iterator i = mds_info.begin();
-	i != mds_info.end(); ++i) {
-      encode(i->first, bl);
-      encode(i->second, bl, features);
-    }
-    n = data_pools.size();
-    encode(n, bl);
-    for (const auto p: data_pools) {
-      n = p;
-      encode(n, bl);
-    }
 
-    int32_t m = cas_pool;
-    encode(m, bl);
-    return;
-  } else if ((features & CEPH_FEATURE_MDSENC) == 0) {
-    __u16 v = 3;
-    encode(v, bl);
+  __u8 v1, v2;
+  __u16 ev;
+  if (features & CEPHFS_FEATURE_OCTOPUS) {
```

Comment:
you can't use CEPHFS_FEATURE_MAX here.  features are from src/include/ceph_features.h

Reply:
I'm not sure which constant to be used for the version beyond octopus. I had checked ceph_features.h file earlier. But I didn't find anything related to the P release you have mentioned in the IRC. Could you please clarify @ukernel ?

Reply:
feature bit for P release hasn't defined.